Mage.Sets/src/mage/cards/t/TrespassersCurse.java

```diff
@@ -100,7 +100,8 @@ public boolean checkTrigger(GameEvent event, Game game) {
         Permanent enchantment = game.getPermanent(this.getSourceId());
         if (enchantment != null
                 && enchantment.getAttachedTo() != null
-                && game.getControllerId(event.getSourceId()).equals(enchantment.getAttachedTo())) {
+                && game.getControllerId(event.getSourceId()).equals(enchantment.getAttachedTo())
+                && game.getPermanent(this.getSourceId()).isCreature()) {
```

Comment:
It's probably karma for not back-filling with unit tests. I work for a TDD-focused company! I should have known :)